lib/providers/view_model/wallet_detail/utxo_detail_view_model.dart

```diff
@@ -34,6 +43,41 @@ class UtxoDetailViewModel extends ChangeNotifier {
     _dateString = DateTimeUtil.formatTimestamp(_transaction!.timestamp);
 
     _initUtxoInOutputList();
+    _walletProvider.addWalletUpdateListener(_walletId, _onWalletUpdate);
+  }
+
+  void _onWalletUpdate(WalletUpdateInfo info) {
+    final updatedUtxo = _walletProvider.getUtxoState(_walletId, _utxoId);
+    if (updatedUtxo != null) {
+      _utxo = updatedUtxo;
+      notifyListeners();
+    }
+  }
+
+  UtxoStatus _setUtxoStateToggled() {
+    if (_utxo.status == UtxoStatus.locked) {
+      return UtxoStatus.unspent;
+    } else if (_utxo.status == UtxoStatus.unspent) {
+      return UtxoStatus.locked;
+    }
+    return _utxo.status;
+  }
+
+  void toggleUtxoLockStatus() async {
+    _utxo = UtxoState(
+      transactionHash: _utxo.transactionHash,
+      index: _utxo.index,
+      amount: _utxo.amount,
+      derivationPath: _utxo.derivationPath,
+      blockHeight: _utxo.blockHeight,
+      to: _utxo.to,
+      timestamp: _utxo.timestamp,
+      status: _setUtxoStateToggled(),
+      spentByTransactionHash: _utxo.spentByTransactionHash,
+    );
+
+    notifyListeners();
+    await _walletProvider.toggleUtxoLockStatus(_walletId, _utxo.index, _utxo.transactionHash);
   }
```

Comment:
```suggestion
await _walletProvider.toggleUtxoLockStatus(_walletId, _utxo.index, _utxo.utxoId);
```

`await _walletProvider.toggleUtxoLockStatus(_walletId, _utxo.utxoId);`

이런 방식으로 더 단순하게 바꾸면 어떨까요?